cats/cats-sql/src/main/kotlin/com/netflix/spinnaker/cats/sql/cache/SqlCache.kt

```diff
@@ -1286,16 +1513,79 @@ class SqlCache(
     }
   }
 
-  private fun selectIdentifiers(type: String, ids: List<String>): MutableCollection<String> {
+  /**
+   * Returns the ids stored for the given type and hashed ids.
+   *
+   * @param type the type of the ids to retrieve.
+   * @param hashedIds the hashed ids of the ids to retrieve.
+   * @return list of ids stored in the cache that matched the type and hashed ids provided.
+   */
+  private fun selectIdentifiers(type: String, hashedIds: List<String>): MutableCollection<String> {
     return withRetry(RetryCategory.READ) {
       jooq.select(field("id"))
         .from(table(sqlNames.resourceTableName(type)))
-        .where(field("id").`in`(*ids.toTypedArray()))
+        .where(field("id_hash").`in`(*hashedIds.toTypedArray()))
         .fetch()
         .intoSet(field("id"), String::class.java)
     }
   }
 
+  /**
+   * Evicts cache records, but does not evict relationship rows. Difference between internal and
+   * public function is that the internal function takes in hashed ids rather than the ids.
+   *
+   * @param type the type of the records that will be removed.
+   * @param hashedIds collection of hashed ids to be removed.
+   */
+  private fun evictAllInternal(type: String, hashedIds: Collection<String>) {
+    if (hashedIds.isEmpty()) {
+      return
+    }
+
+    log.info("evicting ${hashedIds.size} $type records")
+
+    var deletedCount = 0
+    var opCount = 0
+    try {
+      hashedIds.chunked(
+        dynamicConfigService.getConfig(
+          Int::class.java,
+          "sql.cache.read-batch-size",
```

Comment:
Wouldn't this want to use `write-batch-size`?  (Feels like delete is closer to write, in the absence of a separate `delete-batch-size` config.)